### PR TITLE
feat: [DHIS2-16262] Support Custom Layout for Enrollment Pages

### PR DESCRIPTION
**Pull request by eirikhaugstulen**
Support building a custom enrollment dashboard layout, using a custom config stored in the data store.
The system administrator should be able to design, reorder, hide and lock a dashboard for a program.

## Overview

Each enrollment page has a separate key and config in the data store:
```js
ENROLLMENT_OVERVIEW: 'enrollmentOverviewLayout',
ENROLLMENT_EVENT_NEW: 'enrollmentEventNewLayout',
ENROLLMENT_EVENT_EDIT: 'enrollmentEventEditLayout',
```

We will check this key for a custom page layout, or fallback to the default one stored under `DefaultPageLayout` in each page.

Some components will also have additional settings that can be overridden by the config.
```ts
// ProfileWidget
settings = {
    readOnlyMode: boolean
}

// EnrollmentWidget
settings = {
    readOnlyMode: boolean
}
```

## Supported components
```ts
type DefaultComponents = 'QuickActions'
    | 'StagesAndEvents'
    | 'AssigneeWidget'
    | 'NewEventWorkspace'
    | 'EditEventWorkspace'
    | 'EnrollmentComment'
    | 'EventComment'
    | 'TrackedEntityRelationship'
    | 'ErrorWidget'
    | 'WarningWidget'
    | 'FeedbackWidget'
    | 'IndicatorWidget'
    | 'ProfileWidget'
    | 'EnrollmentWidget';
```

## Examples
Custom config to build the default enrollment dashboard (Child Programme):
```json
{
  "IpHINAT79UW": {
    "title": "Child Programme dashboard"
    "leftColumn": [
      {
        "type": "component",
        "name": "QuickActions"
      },
      {
        "type": "component",
        "name": "StagesAndEvents"
      }
    ],
    "rightColumn": [
      {
        "type": "component",
        "name": "TrackedEntityRelationship"
      },
      {
        "type": "component",
        "name": "ErrorWidget"
      },
      {
        "type": "component",
        "name": "WarningWidget"
      },
      {
        "type": "component",
        "name": "FeedbackWidget"
      },
      {
        "type": "component",
        "name": "IndicatorWidget"
      },
      {
        "type": "component",
        "name": "Notes"
      },
      {
        "type": "component",
        "name": "ProfileWidget",
        "settings": {
             "readOnlyMode": true
        }
      },
      {
        "type": "component",
        "name": "EnrollmentWidget",
         "settings": {
             "readOnlyMode": true
        }
      }
    ]
  }
}
```

## Optional parameters
All parts of the config is optional. You can choose to:
- Not have a title above the page
- Only have one column for the full page
- Place any component in any column (Given it's supported by the page)

## Images
Example:
image

### CODE REVIEW

File: src/core_modules/capture-core/components/Pages/common/EnrollmentOverviewDomain/EnrollmentPageLayout/hooks/useWidgetColumns.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    props: Object,
+};
+
+const MemoizedWidgets: { [key: string]: React$ComponentType<any> } = {};
+const UnsupportedWidgets: { [key: string]: boolean } = {};
+
+const renderWidget = (widget: ColumnConfig, availableWidgets, props) => {
+    const { type } = widget;
+
+    if (type.toLowerCase() === WidgetTypes.COMPONENT) {
```
Comment (simonadomnisoru) [inline src/core_modules/capture-core/components/Pages/common/EnrollmentOverviewDomain/EnrollmentPageLayout/hooks/useWidgetColumns.js:21]: Are we going to support other types of widgets in the future?

**Review by simonadomnisoru — APPROVED** [review]
Great job on this feature 👏 ! The code is very well structured and easy to follow

File: src/core_modules/capture-core/components/Pages/common/EnrollmentOverviewDomain/EnrollmentPageLayout/hooks/useWidgetColumns.js (hunk 1 above)
Comment (eirikhaugstulen) [reply src/core_modules/capture-core/components/Pages/common/EnrollmentOverviewDomain/EnrollmentPageLayout/hooks/useWidgetColumns.js:21]: Thanks for the review! Yes, we will extend this config to also support Enrollment Dashboard plugins. 😊
https://dhis2.atlassian.net/browse/DHIS2-16305

**Review by geethaalwan — APPROVED** [review]
Tested successfully on 2.41,2.40.3,2.39.5,2.38.7 versions

**Pull request merged**